lib/companies/job_checker.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Companies.JobChecker do
+  alias Companies.Jobs
+
+  def check_job(job) do
+    case job.url |> HTTPoison.head() do
+      {:ok, _response = %HTTPoison.Response{status_code: 404}} ->
+        Jobs.delete(job)
+
+      {:ok, %HTTPoison.Response{}} ->
+        nil
+
+      {:error, _} ->
+        nil
+    end
+  end
+
+  def check_all do
+    Jobs.all() |> Enum.each(&check_job(&1))
```

Comment:
Could we change this to:

```suggestion
Enum.each(Jobs.all(), &check_job(&1))
```

https://github.com/christopheradams/elixir_style_guide#avoid-single-pipelines